torchrl/modules/tensordict_module/rnn.py

```diff
@@ -0,0 +1,234 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Optional, Tuple
+
+import torch
+
+from tensordict.nn import TensorDictModuleBase as ModuleBase
+
+from tensordict.tensordict import NO_DEFAULT, TensorDictBase
+from tensordict.utils import prod
+
+from torch import nn
+
+
+class LSTMModule(ModuleBase):
+    """An embedder for an LSTM module.
+
+    This class adds the following functionality to :class:`torch.nn.LSTM`:
+
+    - Compatibility with TensorDict: the hidden states are reshaped to match
+      the tensordict batch size.
+    - Optional multi-step execution: with torch.nn, one has to choose between
+      :class:`torch.nn.LSTMCell` and :class:`torch.nn.LSTM`, the former being
+      compatible with single step inputs and the latter being compatible with
+      multi-step. This class enables both usages.
+
+
+    After construction, the module is *not* set in temporal mode, ie. it will
+    expect single steps inputs.
+
+    If in temporal mode, it is expected that the last dimension of the tensordict
+    marks the number of steps. There is no constrain on the dimensionality of the
+    tensordict (except that it must be greater than one for temporal inputs).
+
+    Args:
+        lstm (torch.nn.Module): a :class:`torch.nn.LSTM` instance. For simplicity, this class does not
+            handle the construction of the module.
+        in_keys (list of str): a triplet of strings corresponding to the input value,
+            first and second hidden key.
+        out_keys (list of str): a triplet of strings corresponding to the output value,
+            first and second hidden key.
+            .. note::
+              For a better integration with TorchRL's environments, the best naming
+              for the output hidden key is ``("next", "hidden0")`` or similar, such
+              that the hidden values are passed from step to step during a rollout.
```

Comment:
Why are we asking the keys like this?

In my mind i would have 3 key arguments:
- in_keys # in keys apart from hidden state (not constrained to 1)
- out_keys # out keys apart from hidden state (not constrained to 1)
- hidden_keys = (hidden0, hidden1) # this is by default like this and the init automatically adds these as input keys and the version of these with preappended next to the output keys